frontend/lib/framework/Context.cpp

```diff
@@ -994,6 +1036,10 @@ bool Context::queryCanUseSavedResult(
   } else if (this->currentRevisionNumber == resultEntry->lastChecked) {
     // the query was already checked/run in this revision
     useSaved = true;
+  } else if (resultEntry->recursionErrors.size()) {
+    // If the query had recursion, don't re-use its result
+    // in subsequent generations.
```

Comment:
This is fine but I think it'd be nice to expand on the comment to say why not. Is the reason just to avoid having to get the reuse logic to be correct in this case? Or maybe because we need to distinguish between failure due to recursion and a query actually returning a default-constructed result?

Reply:
It's a little bit of both. I've added a comment.